File: Triatlon-C#/TriatlonNetworking/rpProtocols/Response.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TriatlonNetworking.rpProtocols
{
    [Serializable]
    public class Response
    {
        public ResponseType type { get; private set; }

        public object data { get; private set; }

        public override string ToString()
        {
            return "Response{" +
                "type='" + type + '\'' +
                ", data='" + data + '\'' +
                '}';
        }

        public class Builder
        {
            private Response response = new Response();

            public Builder type(ResponseType type)
            {
                response.type = type;
                return this;
            }

            public Builder data(object data)
            {
                response.data = data;
                return this;
            }

            public Response build()
            {
                return response;
            }
        }
    }
}

File: Triatlon-Java/TriatlonNetworking/src/main/java/rpcprotocol/Response.java
package rpcprotocol;

import java.io.Serializable;

public class Response implements Serializable {
private ResponseType type;
    private Object data;

    private Response() {
    }

    public ResponseType type() {
        return type;
    }

    public Object data() {
        return data;
    }

    @Override
    public String toString() {
        return "Response{" +
                "type='" + type + '\'' +
                ", data='" + data + '\'' +
                '}';
    }

    public static class Builder {
        private Response response = new Response();

        public Builder type(ResponseType type) {
            response.type(type);
            return this;
        }

        public Builder data(Object data) {
            response.data(data);
            return this;
        }

        public Response build() {
            return response;
        }
    }

    private void data(Object data) {
        this.data = data;
    }

    private void type(ResponseType type) {
        this.type = type;
    }

}
